ext/ext.go

```diff
@@ -0,0 +1,160 @@
+package ext
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"sort"
+	"strings"
+	"sync"
+)
+
+// TODO: Make an ExtensionRegistry?
+//
+//nolint:gochecknoglobals
+var (
+	mx         sync.RWMutex
+	extensions = make(map[ExtensionType]map[string]*Extension)
+)
+
+// ExtensionType is the type of all supported k6 extensions.
+type ExtensionType uint8
+
+// All supported k6 extension types.
+const (
+	JSExtension ExtensionType = iota + 1
+	OutputExtension
+)
+
+func (e ExtensionType) String() string {
+	var s string
+	switch e {
+	case JSExtension:
+		s = "js"
+	case OutputExtension:
+		s = "out"
+	}
+	return s
+}
+
+// Extension is a generic container for any k6 extension.
+type Extension struct {
+	Name, Path, Version string
+	Type                ExtensionType
+	Mod                 interface{}
+}
+
+func (e Extension) String() string {
+	return fmt.Sprintf("%s %s, %s [%s]", e.Path, e.Version, e.Name, e.Type)
+}
+
+// Register a new extension with the given name and type. This function will
+// panic if an unsupported extension type is provided, or if an extension of the
+// same type and name is already registered.
+func Register(name string, typ ExtensionType, mod interface{}) {
+	mx.Lock()
+	defer mx.Unlock()
+
+	exts, ok := extensions[typ]
+	if !ok {
+		panic(fmt.Sprintf("unsupported extension type: %T", typ))
+	}
+
+	if _, ok := exts[name]; ok {
+		panic(fmt.Sprintf("extension already registered: %s", name))
+	}
+
+	path, version := getModuleInfo(mod)
+
+	exts[name] = &Extension{
+		Name:    name,
+		Type:    typ,
+		Mod:     mod,
+		Path:    path,
+		Version: version,
+	}
+}
+
+// Get returns all extensions of the specified type.
+func Get(typ ExtensionType) map[string]*Extension {
+	mx.RLock()
+	defer mx.RUnlock()
+
+	exts, ok := extensions[typ]
+	if !ok {
+		panic(fmt.Sprintf("unsupported extension type: %T", typ))
+	}
+
+	result := make(map[string]*Extension, len(exts))
+
+	for name, ext := range exts {
+		result[name] = ext
+	}
+
+	return result
+}
+
+// GetAll returns all extensions, sorted by their import path and name.
+func GetAll() []*Extension {
+	mx.RLock()
+	defer mx.RUnlock()
+
+	js, out := extensions[JSExtension], extensions[OutputExtension]
+	result := make([]*Extension, 0, len(js)+len(out))
+
+	for _, e := range js {
+		result = append(result, e)
+	}
+	for _, e := range out {
+		result = append(result, e)
+	}
+
+	sort.Slice(result, func(i, j int) bool {
+		if result[i].Path == result[j].Path {
+			return result[i].Name < result[j].Name
+		}
+		return result[i].Path < result[j].Path
+	})
+
+	return result
+}
+
+// getModuleInfo attempts to return the package path and version of the Go
+// module that created the given value.
+func getModuleInfo(mod interface{}) (path, version string) {
```

Comment:
Maybe `extractModuleInfo` ?

Reply:
What's wrong with `get*`? You really don't seem to like that word :smile: 

Sure, "extract" could be used here as well, but "get" is a standard name, and is clear enough IMO. :man_shrugging:

Reply:
No, I'm not against the `get*`, but the truth is that it's too generic, and if it's possible to provide a more precise name, it will be better since it saves time for a code reader.

I have a feeling that extract can tell more to the end user than the `get`, since, from the method name user already can understand that we're extracting details from the first argument without jumping to the method body.

Reply:
I see, that makes sense. I renamed it to `extractModuleInfo` in b1f3c94b6.